### PR TITLE
xmrig 2.2.1

### PR DESCRIPTION
**Pull request by rtau**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/xmrig.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Xmrig < Formula
+  desc ""
```
Comment (fxcoudert) [inline Formula/xmrig.rb]: Description missing

File: Formula/xmrig.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  homepage "https://github.com/xmrig/xmrig"
+  url "https://github.com/xmrig/xmrig.git", :tag => "v2.2.1", :revision => "92d787c817dfa92ca38e36f634195f31ff7d40c3"
```
Comment (fxcoudert) [inline Formula/xmrig.rb]: We prefer building from source tarball (`https://github.com/xmrig/xmrig/archive/v2.2.1.tar.gz`) than git tag

File: Formula/xmrig.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+  depends_on "cmake" => :build
+  depends_on "libuv"
+
+  def install
+    Dir.mkdir("build")
+    Dir.chdir("build")
+    system "cmake", "..", "-DCMAKE_BUILD_TYPE=Release", "-DUV_LIBRARY=/usr/local/opt/libuv/lib/libuv.a", *std_cmake_args
```
Comment (fxcoudert) [inline Formula/xmrig.rb]: For path to static libuv, please use: `#{Formula["libuv"].opt_lib}/libuv.a`

File: Formula/xmrig.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+class Xmrig < Formula
+  desc ""
+  homepage "https://github.com/xmrig/xmrig"
+  url "https://github.com/xmrig/xmrig.git", :tag => "v2.2.1", :revision => "92d787c817dfa92ca38e36f634195f31ff7d40c3"
+
+  depends_on "cmake" => :build
+  depends_on "libuv"
+
+  def install
+    Dir.mkdir("build")
```
Comment (fxcoudert) [inline Formula/xmrig.rb]: `mkdir "build" do`

File: Formula/xmrig.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    Dir.chdir("build")
+    system "cmake", "..", "-DCMAKE_BUILD_TYPE=Release", "-DUV_LIBRARY=/usr/local/opt/libuv/lib/libuv.a", *std_cmake_args
+    system "make"
+    bin.install "xmrig"
```
Comment (fxcoudert) [inline Formula/xmrig.rb]: No `make install`? Nothing other than the `xmrig` binary to install? Man pages, etc.?

File: Formula/xmrig.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  test do
+    system "#{bin}/xmrig", "-V"
```
Comment (fxcoudert) [inline Formula/xmrig.rb]: We need a test that does something, no just output version number

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by ilovezfs** [comment]
@rtau thanks for the pull request. Please feel free to open a new one when the feedback has been addressed.

**Pull request closed without merging**